### PR TITLE
Update index.rst

### PR DESCRIPTION
**Pull request by samanthahamilton**
Summary: Changes as per the DOCX Mockup file.

Hi Ian,

First, wow, you've made so many amazing changes over the weekend! Renaming PR https://github.com/ianhi/mpl-interactions/pull/46 to "radical documentation improvements" was so true!

I've taken the word doc and made the appropriate changes to the index "landing page." Here are some additional comments:

- I agree that mpl_interactions is the better version, stylistically and technically. I've made that adjustment (in addition to the others) to my style sheet.
- Your comments about when to use periods with bullet points, and parallel structure, were kind, but that's not your job, that's mine! You code, and I'll edit!
- I think this "version" of the landing page really supports your primary goals: the audience is addressed, the language is simple, majority of the key information is within one page, and there is direction to further/deeper information available on other pages.
- I was not able to build the documentation in HTML to test before sending this PR. I received this message:

`Configuration error: There is a programmable error in your configuration file: Traceback (most recent call last):`
`  File "c:\python38\lib\site-packages\sphinx\config.py", line 319, in eval_config_file`
`    execfile_(filename, namespace)`
`  File "c:\python38\lib\site-packages\sphinx\util\pycompat.py", line 89, in execfile_`
`    exec(code, _globals)`
`  File "C:\Users\colek\Documents\GitHub\mpl-interactions\docs\conf.py", line 18, in <module>`
`    release = mpl_inter.__version__`
`AttributeError: module 'mpl_interactions' has no attribute '__version__'` 

Now that the documentation skeleton has really been developed, did you want to close Issue https://github.com/ianhi/mpl-interactions/issues/35 and create a new issue/new branch? Perhaps something like **docsediting**?

Let me know if this was helpful, or if there's anything else to dive into,
Cheers,
Sam.

### CODE REVIEW

**Comment by samanthahamilton** [comment]
Sorry, @ianhi, In my continued education on GitHub contributions, I realize now I probably should have created a new branch for the Contribution page suggested changes? Moving forward, what would be easiest for you & the docs? Would separate branches for separate pages be better for your workflow? Or is "documentation" all one concept for you?

To address the new update re `Contributing.rst`:

- I made some brief grammatical changes throughout (this page was really well presented!)
- I added a draft table for a Contributors section, like [here](https://covid19-scenarios.org/team)

Talk soon,
Sam.

**Comment by ianhi** [comment]
I'm trapped in day of zoom meetings :( so I'll be quick with a few comments

> I was not able to build the documentation in HTML to test before sending this PR. I received this message:

How did you install the library? The `__version__` was part of that massive scope creep so it's likely that you don't have the most up to date version install. You should try installing like like: `pip install -e .[docs]` the `-e` is important, it means editable, and will use the files from the git repo as the files it loads the library from.


**new branches**

I think keeping it here in on PR is totally reasonable. There is a consistent theme behind the commits in that they are all documentaiton based. So I think it's better actually to keep them all here. 


> I added a draft table for a Contributors section, like here

Great idea! Though I think I prefer the all-contributors system as it may be easier to manage. https://github.com/all-contributors/all-contributors

I added you for `docs` see https://github.com/ianhi/mpl-interactions/pull/67. So if you're cool with that let me know and ill merge that
see results here https://github.com/ianhi/mpl-interactions#contributors-

File: docs/index.rst [hunk 1]
```diff
@@ -1,89 +1,56 @@
...
+   fig, ax, sliders = interactive_plot([f1, f2], x=x, τ = τ, β = β)
+   _ = plt.legend()
 
-   Installation
-   Backends
-   compare-to-ipywidgets
-   API
-   Examples
-   Contributing
-
```
Comment (ianhi) [inline docs/index.rst:79]: We need to keep this toctree otherwise we lose the contents of the navigation sidebar

Comment (ianhi) [reply docs/index.rst:79]: you can just add it back in below the example

**Comment by ianhi** [comment]
oh also

> or if there's anything else to dive into,

Anything and everything you feel inspired to look at would be awesome. In particular the `Examples` page and the docstrings of the functions like this one https://github.com/ianhi/mpl-interactions/blob/8b9e2762d323f88205113519f83c41443190f684/mpl_interactions/jupyter.py#L194-L252

docstrings are little pieces of info that pop up while you're coding to remind you how a function works, so they are the piece of documentation that is most frequently encountered by a user. I have been trying to follow the numpydoc style https://numpydoc.readthedocs.io/en/latest/format.html

as numpy is really the foundational scientific computing package in python

**Comment by ianhi** [comment]
and then those docstrings are used to autogenerate pages on the docs website. So here is the rendered version of the above docstring https://mpl-interactions.readthedocs.io/en/latest/autoapi/mpl_interactions.interactive_plot.html#mpl_interactions.interactive_plot

**Comment by samanthahamilton** [comment]
Hi Ian!

I hope all those Zoom meetings went well. Good news! I figured out what went wrong with rendering the HTML build. Python lost/stopped seeing/couldn't find `pip`, but I figured it all out (learned how to make a batch file).

Here are some edits to the `/index` and `/contributors pages`, as per your notes.

- I added the toc tree back, but kept a buffer line of text between the bottom example and the navigation (for a visual space and readable transition)
- I think the all-contributors approach is great addition! It looks very professional

> There is a consistent theme behind the commits in that they are all documentation based. So I think it's better actually to keep them all here.

I will keep the Issue35-documentation branch running, and continue to work through and edit the docs within the same issue. 

I think these two pages are really looking and sounding professional---it's very exciting to see things coming together.

I have some vacation coming up, and I will work through the `/examples` page, and some general edits throughout. Best, Sam.

**Comment by ianhi** [comment]
@samanthahamilton  two pieces of python/git advice

> Python lost/stopped seeing/couldn't find pip, but I figured it all out (learned how to make a batch file).

yikes thats the worst. I probably should have recommeneded this earlier: Managing your python environments with anaconda (https://www.anaconda.com/products/individual#windows) is probably the best way to manage a python installation for scientific software. Congrats on figuring it out though! If what you have works I would keep doing that until it breaks again and only then try to switch to conda.

> I will keep the Issue35-documentation branch running, and continue to work through and edit the docs within the same issue.

You need to make sure to update that branch with the changes that get made to the master branch. For example there have already been a few changes via other commits/PRs that are in the master branch but not your branch because you made it before I added those other commits. You can do a few things here but all of them require that you first get the changes on this repos master branch into your repos master branch. I made a GIF of how to do that with Github desktop using my fork of the matplotlib repo:

how-to-merge-master

After you do that you can
1. Make a new branch from the updated master (this is what I do)
2. rebase your branch on the master branch `git rebase master`
3. Merge your master branch into your issue-35 branch

I'd recommend doing the first one as I find it to be simplest. (also to keep this process easy you should make sure to never commit directly to your master branch)


> I have some vacation coming up

enjoy :tada:

**Comment by samanthahamilton** [comment]
Hey @ianhi,

I know this PR is now merged (yay!), but I was wondering if I could pick your brain about Git/Github? First, a big thanks for the gif! I've never merged that way before. This is my "new/current" workflow. I'd love to know if it makes sense, or if there are suggestions you would have? It's mostly a concoction based on reading and watching a thousand articles, tutorials, and videos...

- Fork > Clone > Create local branch > open correct files > make changes > save into GitHub desktop > Commit > Push to Chrome > Publish the branch > create a Pull Request > it's merged by the maintainer 

This ^ basic workflow is straightforward to me, and if I make the fork, clone, changes, PR, in one "sitting" usually nothing has changed in the upstream master branch. BUT when updates to the actual repo are made and I get behind, this is what I _think_ happens:

1. A previously forked repo exists as samanthahamilton/X, a previously created clone of this "remote origin" exists locally, and a branch I'm working on exists locally, but now these are all behind...
2. Open GitHub desktop > select the master branch > view files in Explorer
3. Open command line
4. Make sure upstream is added: `git remote add upstream http://github.com/project-name.git`
5. `git pull upstream` Which _I believe_ fetches and merges any changes from the upstream repo into my local master branch, and _I think_ pulls for all branches? I read somewhere that if it was `git pull upstream/master` it would only update that branch, but leaving it blank pulls all?
6. `git push origin` Which _I believe_ then pushes these updates into my remote origin (samanthahamilton/X) to make my fork even with the master
7. Then I usually enter `git status` for fun
8. Then, in GitHub Desktop, I select my working branch from the dropdown list, and keep working...

Thank you for being my GitHub Guru!
Sam.

**Comment by ianhi** [comment]
> Open command line

You can do all of git through the command line as it is first and foremost a command line program. But programs like github desktop add a nice gui. Alternatively for this whole list you can do it all in github desktop. You should do whatever you find easiest.


> This is what I have been doing and it's probably wrong git fetch upstream (no message) then git rebase upstream/master (message: successfully rebased and updated refs/head/master). But I think rebasing is wrong? I read somewhere about rewriting the history...

Yah I don't think this is the optimal strategy. Though it doesn't seem to have been a problem. Still, I think merging is the more correct thing for you to be doing here as rebase and merge are fundamentally different. See https://www.atlassian.com/git/tutorials/merging-vs-rebasing

> git remote add upstream http://github.com/project-name.git

you only need to do this once. In the git repo there's a hidden folder called `.git` and if you look at the file in there `config` you should see all the remotes you've added - probably `origin` and `upstream`. 

> git push origin Which I believe then pushes these updates into my remote origin (samanthahamilton/X) to make my fork even with the master

Indeed. Though a small note on language. This makes your fork's master branch equal to my repo's master branch.

**Pull request merged**